sklearn/utils/estimator_checks.py

```diff
@@ -521,7 +521,7 @@ def check_estimator(Estimator, generate_only=False, strict_mode=True):
 
     Parameters
     ----------
-    estimator : estimator object
```

Comment:
actually it will not be possible to pass a class from 0.24, see [L560](https://github.com/scikit-learn/scikit-learn/pull/18599/files#diff-a95fe0e40350c536a5e303e87ac979c4L560) so I am not sure that we want to change this because I believe that we will change the `Estimator` parameter to `estimator`.

Reply:
This is slightly annoying because indeed we switched from classes to instances, and we didn't want to deprecate `Estimator` for `estimator`.

Documenting `estimator` instead of `Estimator` clarifies that we expect an instance, but indeed the documented name is inconsistent with the actual parameter name. I'm not sure which is best here.

Reply:
What about changing to
```python
@_deprecate_positional_args
def check_estimator(estimator=None, *, generate_only=False, strict_mode=True, Estimator=None):
```
and also throwing a deprecation warning if  `Estimator=something` was set?
Just an idea how to get it right in the future.

Reply:
That would be deprecating `Estimator` (regardless of the use of `_deprecate_positional_args`), which we wanted to avoid / deemed not worth it.

We can deprecate if you think it's cleaner (another PR though). I doubt anyone actually uses `check_estimator(Estimator=...)` anyway, so hopefully this shouldn't be a problem for any third party library